peers/app_request_network.go

```diff
@@ -351,52 +360,121 @@ func (n *appRequestNetwork) trackSubnet(subnetID ids.ID) bool {
 func (n *appRequestNetwork) TrackSubnet(ctx context.Context, subnetID ids.ID) {
 	// Track the subnet. Update the validator set if we weren't already tracking it.
 	if !n.trackSubnet(subnetID) {
-		n.updateValidatorSet(ctx, subnetID)
+		n.updateTrackedValidatorSet(ctx, subnetID)
 	}
 }
 
-func (n *appRequestNetwork) startUpdateValidators(ctx context.Context) {
+func (n *appRequestNetwork) startUpdateTrackedValidators(ctx context.Context) {
 	// Fetch validators immediately when called, and refresh every ValidatorRefreshPeriod
 	ticker := time.NewTicker(ValidatorRefreshPeriod)
+
 	for {
 		select {
 		case <-ticker.C:
-			n.trackedSubnetsLock.RLock()
-			subnets := n.trackedSubnets.List()
-			n.trackedSubnetsLock.RUnlock()
```

Comment:
Looks like we removed this protection here but didn't add it again when calling `n.trackedSubnets.List()` inside `n.updateTrackedValidatorSets`